### PR TITLE
Topic/ta14 018

### PR DESCRIPTION
**Pull request by godunko**

### CODE REVIEW

File: source/spawn/spawn-processes-windows.adb [hunk 1]
```diff
@@ -239,56 +244,178 @@ package body Spawn.Processes.Windows is
...
-            Success := False;
-            return;
-         end if;
+         --  Create named pipe and server handle
+
+         loop
+            declare
+               Pipe_Count_Image : constant String :=
+                 Interfaces.Unsigned_32'Image (Pipe_Count);
+               --  Pipe_Name        : constant Interfaces.C.char_array :=
```
Comment (setton) [inline source/spawn/spawn-processes-windows.adb:285]: commented out code, should be removed

File: source/spawn/spawn-processes-windows.adb [hunk 2]
```diff
@@ -239,56 +244,178 @@ package body Spawn.Processes.Windows is
...
+                     Parent_Handle := System.Win32.INVALID_HANDLE_VALUE;
+                     Success       := False;
+
+                     return;
+                  end if;
+               end;
+            end;
+         end loop;
+
+         --  Open named pipe to create clinet handle
```
Comment (setton) [inline source/spawn/spawn-processes-windows.adb:341]: *client

**Review by setton — CHANGES_REQUESTED** [review]

File: source/spawn/spawn-processes-windows.adb [hunk 3]
```diff
@@ -239,56 +244,176 @@ package body Spawn.Processes.Windows is
...
-            dwFlagsAndAttributes  => System.Win32.FILE_ATTRIBUTE_NORMAL,
+            dwFlagsAndAttributes  => Windows_API.FILE_FLAG_OVERLAPPED,
             hTemplateFile         => 0);
 
          if Child_Handle = System.Win32.INVALID_HANDLE_VALUE then
             Self.Listener.Error_Occurred (Integer (System.Win32.GetLastError));
-            Success := False;
+
+            Interfaces.C.Strings.Free (Pipe_Name);
+            Ignore (System.Win32.CloseHandle (Parent_Handle));
```
Comment (setton) [inline source/spawn/spawn-processes-windows.adb:354]: I feel like we shouldn't ignore the result here and emit a trace in case it's False

**Review by setton — APPROVED** [review]

File: source/spawn/spawn-processes-windows.adb (hunk 3 above)
Comment (godunko) [reply source/spawn/spawn-processes-windows.adb:354]: It is not necessary here, handle is valid one and not created by some specific functions which requires specific cleanup

**Pull request merged**